sys/compat/linuxkpi/common/include/linux/xarray.h

```diff
@@ -122,25 +122,25 @@ xa_init(struct xarray *xa)
 }
 
 static inline void *
-xa_mk_value(unsigned long v)
+xa_mk_value(uintptr_t v)
```

Comment:
Shouldn't the interface remain unsigned long? This can't actually encode/decode pointers.

Reply:
It returns 'void *' so I am not sure how to convert unsigned long to it (-Wcheri-capability-misuse)

Reply:
With a cast via uintptr_t

Reply:
I had made it uintptr_t originally because I wasn't sure if existing code might pass pointers to this, but maybe that's some other xa_mk_* routine we don't currently have a shim for?  This version is fine with me though (with it reverted back to unsigned long).